sunbeam-python/sunbeam/plugins/validation/plugin.py

```diff
@@ -0,0 +1,484 @@
+        with console.status(f"Retrieving {TEMPEST_APP_NAME}'s unit name."):
+            app = TEMPEST_APP_NAME
+            model = OPENSTACK_MODEL
+            unit = run_sync(jhelper.get_leader_unit(app, model))
+            if not unit:
+                message = f"Unable to get {app} leader"
+                raise click.ClickException(message)
+            return unit
+
+    def _run_action_on_tempest_unit(
+        self,
+        action_name: str,
+        action_params: Optional[dict] = None,
+        progress_message: str = "",
+        print_stdout: bool = False,
+    ) -> None:
+        """Run the charm's action."""
+        unit = self._get_tempest_leader_unit()
+        jhelper = JujuHelper(self.client, self.snap.paths.user_data)
+        with console.status(progress_message):
+            action_result = run_sync(
+                jhelper.run_action(
+                    unit,
+                    OPENSTACK_MODEL,
+                    action_name,
+                    action_params or {},
+                )
+            )
+
+            if action_result.get("return-code", 0) > 1:
+                message = f"Unable to run action: {action_name}"
+                raise click.ClickException(message)
+
+            if print_stdout:
+                console.print(action_result.get("summary").strip())
+
+    def _check_file_exist_in_tempest_container(self, filename: str) -> bool:
+        """Check if file exist in tempest container."""
+        unit = self._get_tempest_leader_unit()
+        # Note: this is a workaround to run command to payload container
+        # since python-libjuju does not support such feature. See related
+        # bug: https://github.com/juju/python-libjuju/issues/1029
+        result = subprocess.run(
+            [
+                "juju",
+                "ssh",
+                "--container",
+                TEMPEST_CONTAINER_NAME,
+                unit,
+                "ls",
+                TEMPEST_VALIDATION_RESULT,
+            ],
+            capture_output=True,
+        )
+        if result.returncode != 0:
+            return False
+        return True
+
+    def _copy_file_from_tempest_container(self, source: str, destination: str) -> None:
+        """Copy file from tempest container."""
+        unit = self._get_tempest_leader_unit()
+        progress_message = (
+            f"Copying {source} from "
+            f"{TEMPEST_APP_NAME} ({TEMPEST_CONTAINER_NAME}) "
+            f"to {destination} ..."
+        )
+        with console.status(progress_message):
+            # Note: this is a workaround to run command to payload container
+            # since python-libjuju does not support such feature. See related
+            # bug: https://github.com/juju/python-libjuju/issues/1029
+            if destination == ".":
+                # juju scp does not allow "." as destination
+                destination = Path("./", Path(source).name)
```

Comment:
This is only for `"."`, it's okay if you provide a directory, it will use the source name by default, it's just not working for `"."`.

E.g.
```
"." --> not  wokring
"/a/b/c" --> /a/b/c/<source_name>
```

 I guess there are some issue with python-libjuju about this but I did not dig in

Reply:
interesting, I don't have this issue
```
ubuntu@sunbeam03:~$ sunbeam validation get-last-result -o ./abc/
ubuntu@sunbeam03:~$ ls
abc  ...
```

Reply:
I guess if in that case, we should take your suggestion, and prevent this from even happening: i.e. we limit the output to non-existing file only, what do you think?

Reply:
Fixed!